src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNVectorFieldMapper.java

```diff
@@ -242,6 +242,10 @@ public void parse(ParseContext context) throws IOException {
                                                     "update knn.plugin.enabled setting to true");
         }
 
+        if (KNNSettings.isCircuitBreakerTriggered()) {
```

Comment:
Should there also be a cluster setting like "CircuitBreakerOnTripEnableWriteBlock" so that users have the ability to opt out of the write block?

Reply:
IMO it would keep their performance at risk. Better to put write blocks on knn index. They still have option to take risk by increasing ```knnmemorycircuit_breakerlimit``` percentage

Reply:
> IMO it would keep their performance at risk. Better to put write blocks on knn index and provide alert. They still have option to take risk by increasing `knnmemorycircuit_breakerlimit` percentage

Reply:
That makes sense.